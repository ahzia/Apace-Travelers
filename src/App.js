import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux'; // useSelectors
import { fetchPostsRequestRockets } from './redux/rockets/rockets';
import { fetchPostsRequestMissions } from './redux/missions/missions';
import store from './redux/configureStore';
import NavBar from './components/NavBar';
import Missions from './components/Mission';
import Rockets from './components/Rockets';
import MyProfile from './components/MyProfile';
import Dragons from './components/Dragons';
import './App.css';

const App = () => {
  const [rockets, setRockets] = useState([]);
  const [missions, setMissons] = useState([]);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchPostsRequestMissions());
    dispatch(fetchPostsRequestRockets());
  }, []);

  // const missions = useSelector((state) => state.spaceReducer.missions);
  // const rockets = useSelector((state) => state.spaceReducer.rockets);

  useEffect(() => {
    store.subscribe(() => {
      setRockets(store.getState().rocket.rockets);
      setMissons(store.getState().mission.missions);
    });
  });
  return (
    <div>
      <Router>
        <header>
          <NavBar />
        </header>
        <main>
          <Switch>
            <Route path="/Mission">
              <Missions missions={missions} />
            </Route>
            <Route path="/Dragons">
              <Dragons />
            </Route>
            <Route path="/MyProfile">
              <MyProfile />
            </Route>
            <Route path="/">
              <Rockets rockets={rockets} />
            </Route>
          </Switch>
        </main>
      </Router>
    </div>
  );
};

export default App;
