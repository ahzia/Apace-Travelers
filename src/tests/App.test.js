import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/configureStore';

describe('Component test snapshot', () => {
  it('App renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
