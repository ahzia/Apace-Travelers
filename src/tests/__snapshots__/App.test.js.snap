// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test snapshot App renders correctly 1`] = `
<div>
  <header>
    <nav
      className="mx-5 navbar navbar-expand-lg navbar-light bg-light"
    >
      <span
        className="navbar-brand"
        href=""
      >
        <img
          alt=""
          className="d-inline-block align-top"
          height="30"
          src="planet.png"
          width="30"
        />
         
        Space Travelers' Hub
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="ms-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <a
              aria-current="page"
              className="text-decoration-none text-dark text-decoration-underline"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Rockets
            </a>
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <a
              aria-current={null}
              className="text-decoration-none text-dark"
              href="/Mission"
              onClick={[Function]}
            >
              Mission
            </a>
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <a
              aria-current={null}
              className="text-decoration-none text-dark"
              href="/Dragons"
              onClick={[Function]}
            >
              Dragons
            </a>
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <a
              aria-current={null}
              className="text-decoration-none text-dark"
              href="/MyProfile"
              onClick={[Function]}
            >
              My Profile
            </a>
          </a>
        </div>
      </div>
    </nav>
  </header>
  <main>
    <div
      className="container d-flex flex-column mx-5 my-2"
    />
  </main>
</div>
`;

exports[`Component test snapshot Missions renders correctly 1`] = `
<div
  className="mission__container"
>
  <div
    className="mission_title"
  >
    <ul>
      <li
        className="name title"
      >
        Mission
      </li>
      <li
        className="description title"
      >
        Description
      </li>
      <li
        className="status title"
      >
        Status
      </li>
      <li
        className="loading"
      >
         
      </li>
    </ul>
  </div>
  <div
    className="mission_title"
  />
</div>
`;

exports[`Component test snapshot Profile renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <div>
        <h3
          className="joined__mission__title"
        >
          No Missions joined!
        </h3>
        <div
          className="joined__mission__container"
        />
      </div>
    </div>
    <div
      className="col-md-3"
    >
      <div>
        <h3
          className="joined__mission__title"
        >
          No Rockets Reserved!
        </h3>
        <div
          className="joined__mission__container"
        />
      </div>
    </div>
    <div
      className="col-md-3"
    >
      <div>
        <h3
          className="joined__mission__title"
        >
          No Dragons Reserved!
        </h3>
        <div
          className="joined__mission__container"
        />
      </div>
    </div>
  </div>
</div>
`;
